src/model/wallet.ts

```diff
@@ -365,10 +367,43 @@ export const loadWalletViem = async <S extends Screen>({
   const selectedWallet = getWalletWithAccount(addressToUse);
   const isHardwareWallet = selectedWallet?.type === walletTypes.bluetooth;
 
-  // For now, we'll skip hardware wallet support in viem version
-  // Hardware wallets would need special handling with viem
+  // Handle Ledger hardware wallets
   if (isHardwareWallet) {
```

Comment:
you could negate to 
```suggestion
  if (!isHardwareWallet) {
```

and early return to avoid nesting